### PR TITLE
fix: fix issue with field array values for rest and graphql apis

### LINKED ISSUE #25817: [Bug]: Random invalid values are auto field in header/params tab on refreshing the API pane
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Attached video : 
https://www.loom.com/share/e1810d43d435482e85242396ea1abebc?sid=4019389a-2455-4d8d-94a2-a488aa944ea7

### Steps To Reproduce

1. Create a new app
2. Add a empty Rest API/GraphQL API
3. Add more than 5 entries for key value pair under header/params tab 
4. Create an error fro any key-value pair
5. Refresh the page

### Public Sample App

_No response_

### Environment

Release

### Issue video log

_No response_

### Version

Appsmith v1.9.29-SNAPSHOT

### PR DESCRIPTION
**Pull request by Irongade**
When key value pairs are written within the headers tab for Rest APIs in the pattern below:

```
[
 {key: "hello", value: "world"},
 {key: "", value: ""},
 {key: "hey", value: "there"},
]
```

it causes an incorrect update of the action state, and the new headers look like the pattern below:

```
[
 {key: "hello", value: "world"},
 {key: "hey", value: ""},
 {key: "", value: "there"},
]
```

This happens for Headers, Params, and Body Form Data fields for both REST and GraphQL Apis.

This PR fixes this issue.

Fixes #25817 

- Bug fix (non-breaking change which fixes an issue)

- [ ] Manual
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/sagas/ActionSagas.ts [hunk 1]
```diff
@@ -403,20 +404,24 @@ export function* fetchActionsForPageSaga(
...
+
+    let action: Action = yield select(getAction, actionPayload.payload.id);
     if (!action) throw new Error("Could not find action to update");
 
     if (isAPIAction(action)) {
-      action = transformRestAction(action);
+      // get api action object from redux form
+      const reduxFormAction: ApiAction = yield select(
+        getFormValues(API_EDITOR_FORM_NAME),
+      );
```
Comment (arunvjn) [inline app/client/src/sagas/ActionSagas.ts:421]: Can you help me understand we need to query the form state to make this update on APIs? 
If it is more reliable, should we use it for all actions ?

Comment (Irongade) [reply app/client/src/sagas/ActionSagas.ts:421]: Because only APIs undergo transformation of their data

**Review by arunvjn — APPROVED** [review]

**Review by AmanAgarwal041 — APPROVED** [review]

**Pull request merged**